### PR TITLE
Tests and timing benchmarks for Fortran bindings

### PR DESCRIPTION
**Pull request by h-m0r1**
## Summary

This PR adds comprehensive Fortran tests and timing benchmarks for the sparse-ir-rs Fortran bindings, significantly improving test coverage and providing performance measurement tools.

### New Test Files

Added extensive test suites covering various aspects of the Fortran bindings:

- **`test_analytic_gf`**: Tests for analytic Green's function transformations
- **`test_basis_functions`**: Tests for IR basis functions and properties
- **`test_core`**: Core functionality tests for IR basis creation and manipulation
- **`test_dlr`**: Tests for Discrete Lehmann Representation (DLR) functionality
- **`test_error_handling`**: Error handling and edge case tests
- **`test_memory`**: Memory management and resource cleanup tests
- **`test_multidim`**: Multi-dimensional array handling tests
- **`test_sampling`**: Tests for sampling point selection and validation
- **`test_timing`**: Performance timing benchmarks with configurable parameters

### Test Infrastructure

- **`test_with_rust_capi.sh`**: Generic Fortran compiler test runner (runs ctest)
- **`test_all_compilers.sh`**: Script to run `test_with_rust_capi.sh` with multiple Fortran compilers (gfortran, ifx, ifort)
- **`run_timing_benchmark.sh`**: Unified timing benchmark script that works both locally and with PBS/qsub

### Timing Benchmark Script

The `run_timing_benchmark.sh` script has been enhanced to:

- Support both local execution and PBS/qsub cluster submission
- Include PBS directives template (commented out) for easy cluster usage
- Provide `TEST_DURATION` flag to switch between short (12 runs) and long (66 runs) benchmarks
- Automatically detect PBS environment and adjust behavior accordingly
- Include module command template (commented out) for cluster environments

**Usage:**
- **Local**: Keep PBS directives commented, run with `./run_timing_benchmark.sh`
- **qsub**: Uncomment PBS directives, adjust queue name and module commands, submit with `qsub run_timing_benchmark.sh`

### Updated Files

- Updated `CMakeLists.txt` files to include new test executables
- Enhanced existing test files (`test_ext.f90`, `test_integration.f90`) with additional test cases
- Updated Fortran bindings and implementation files to support new test scenarios
- Modified Python scripts for generating fit/evaluate inc files to declare the input `statistics` correctly:
  - `fortran/script/generate_fit_tau.py`
  - `fortran/script/generate_fit_matsubara.py`
  - `fortran/script/generate_evaluate_tau.py`
  - `fortran/script/generate_evaluate_matsubara.py`
- Regenerated fit/evaluate inc files with corrected input statistics declarations:
  - `fortran/src/fit_tau_impl.inc`
  - `fortran/src/fit_matsubara_impl.inc`
  - `fortran/src/evaluate_tau_impl.inc`
  - `fortran/src/evaluate_matsubara_impl.inc`

## Testing

All new tests have been verified to work with:
- gfortran
- Intel Fortran (ifort)
- Intel Fortran LLVM (ifx)

## Notes

- The timing benchmark script uses environment-specific settings (PBS queue, module commands) that are commented out by default. Users should uncomment and customize these for their cluster environment.

### CODE REVIEW

File: fortran/CMakeLists.txt [hunk 1]
```diff
@@ -140,8 +140,10 @@ add_library(sparseir_fortran SHARED
 if(CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
     # Intel Fortran: Put all arrays on heap to avoid stack overflow
     #target_compile_options(sparseir_fortran PRIVATE -heap-arrays 0)
-    # Intel C++ uses -fp-model=precise to disable fastmath for xprec compatibility
-    # This is handled in the C++ backend CMakeLists.txt
+    # Intel Fortran: Use precise floating-point model (same as Intel C++ for xprec compatibility)
+    target_compile_options(sparseir_fortran PRIVATE -fp-model precise)
+    # Also apply to test executables
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fp-model precise")
```
Comment (shinaoka) [inline fortran/CMakeLists.txt]: Remove -fp-model precise

File: fortran/test/CMakeLists.txt [hunk 2]
```diff
@@ -2,13 +2,31 @@
 set(TEST_FILES
     test_integration
     test_ext
+    test_core
```
Comment (shinaoka) [inline fortran/test/CMakeLists.txt]: Use GLOB to find all test files

File: fortran/CMakeLists.txt (hunk 1 above)
Comment (h-m0r1) [reply fortran/CMakeLists.txt]: It is already commented out not to use the flag.
